cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -287,57 +422,23 @@ func (s *Supervisor) createInstanceID() (ulid.ULID, error) {
 
 }
 
-func keyVal(key, val string) *protobufs.KeyValue {
-	return &protobufs.KeyValue{
-		Key: key,
-		Value: &protobufs.AnyValue{
-			Value: &protobufs.AnyValue_StringValue{StringValue: val},
-		},
-	}
-}
-
-func (s *Supervisor) createAgentDescription() *protobufs.AgentDescription {
-	hostname, _ := os.Hostname()
-
-	return &protobufs.AgentDescription{
-		IdentifyingAttributes: []*protobufs.KeyValue{
-			keyVal("service.name", agentType),
-			keyVal("service.version", s.agentVersion),
-			keyVal("service.instance.id", s.instanceID.String()),
-		},
-		NonIdentifyingAttributes: []*protobufs.KeyValue{
-			keyVal("os.family", runtime.GOOS),
-			keyVal("host.name", hostname),
+func (s *Supervisor) composeExtraLocalConfig() []byte {
+	var cfg bytes.Buffer
+	err := s.extraConfigTemplate.Execute(
```

Comment:
can this get test coverage?

Reply:
I suppose this is as good of a place to start with unit tests as any, I'll add these.

Reply:
I added a basic test to cover composing the effective config, which includes a call to this method.